akka-persistence-testkit/src/main/scala/akka/persistence/testkit/query/scaladsl/PersistenceTestKitReadJournal.scala

```diff
@@ -62,9 +73,30 @@ final class PersistenceTestKitReadJournal(system: ExtendedActorSystem, @unused c
       persistenceId: String,
       fromSequenceNr: Long = 0,
       toSequenceNr: Long = Long.MaxValue): Source[EventEnvelope, NotUsed] = {
-    Source.fromGraph(new EventsByPersistenceIdStage(persistenceId, fromSequenceNr, toSequenceNr, storage))
+    Source
+      .fromGraph(
+        new EventsByPersistenceIdStage[Any](
+          persistenceId,
+          fromSequenceNr,
+          toSequenceNr,
+          storage,
+          persistence.sliceForPersistenceId))
+      .map(env =>
+        EventEnvelope(env.offset, env.persistenceId, env.sequenceNr, env.event, env.timestamp, env.eventMetadata))
```

Comment:
If we change the `EventsByPersistenceIdStage` to emit TimestampOffset we should convert that back to Sequence  here for compatibility.